keras_cv/layers/preprocessing/random_sharpness.py

```diff
@@ -0,0 +1,124 @@
+    original image and the blurred image.  This operation makes the edges of an image
+    less sharp than they were in the original image.
+
+    References:
+        [Sharpness](https://pillow.readthedocs.io/en/stable/reference/ImageEnhance.html)
+
+    Args:
+        factor: Either a tuple of two floats or a single float. `factor` controls the
+            extent to which the image sharpness is impacted.  `factor=0.0` makes this
+            layer perform a no-op operation, while a value of 1.0 uses the sharpened
+            result entirely.  Values between 0 and 1 result in linear interpolation
+            between the original image and the sharpened image.
+
+            Values should be between `0.0` and `1.0`.  If a tuple is used, a `factor` is
+            sampled between the two values for every image augmented.  If a single float
+            is used, a value between `0.0` and the passed float is sampled.  In order to
+            ensure the value is always the same, please pass a tuple with two identical
+            floats: `(0.5, 0.5)`.
+        value_range: the range of values the incoming images will have.
+            Represented as a two number tuple written [low, high].
+            This is typically either `[0, 1]` or `[0, 255]` depending
+            on how your preprocessing pipeline is setup.  Defaults to
+            `[0, 255].`
+    """
+
+    def __init__(
+        self,
+        factor,
+        value_range=(0, 255),
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.value_range = value_range
+        self.factor = RandomSharpness._parse_factor(factor)
+
+    @staticmethod
+    def _parse_factor(factor):
+        if isinstance(factor, float):
+            factor = (0.0, factor)
+
+        if factor[0] > factor[1]:
+            raise ValueError(
+                "`factor[0] > factor[1]`, `factor[0]` must be <= "
+                f"`factor[1]`.  Got `factor={factor}`"
+            )
+        if factor[0] < 0.0 or factor[1] > 1.0:
+            raise ValueError(
+                "`factor` should be inside of range [0.0, 1.0]. " f"Got factor={factor}"
+            )
+
+        return factor
+
+    def get_random_tranformation(self):
+        if isinstance(self.factor, tuple):
+            if self.factor[0] == self.factor[1]:
+                return self.factor[0]
+            return self._random_generator.random_uniform(
+                (), self.factor[0], self.factor[1], dtype=tf.float32
+            )
+
+    def augment_image(self, image, transformation=None):
+        image = preprocessing.transform_value_range(
+            image, original_range=self.value_range, target_range=(0, 255)
+        )
+        original_image = image
+
+        # Make image 4D for conv operation.
+        image = tf.expand_dims(image, axis=0)
+
+        # SMOOTH PIL Kernel.
+        kernel = (
+            tf.constant(
+                [[1, 1, 1], [1, 5, 1], [1, 1, 1]], dtype=tf.float32, shape=[3, 3, 1, 1]
```

Comment:
There are quite some magic number here, eg [1, 5, 1] and (/ 13). I think this will require some documentation about why we does it in this way.

Reply:
Thanks - yeah I added some information.  This might be the first time my traditional signal processing background has come in handy in a modern CV context haha.